Vagrantfile

```diff
@@ -1,35 +1,45 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+require 'fileutils'
+
 netplugin_synced_gopath="/opt/golang"
-host_gobin_path="/opt/go/bin"
-host_goroot_path="/opt/go/root"
+FileUtils.cp "/etc/resolv.conf", Dir.pwd
 
 provision_common = <<SCRIPT
 ## setup the environment file. Export the env-vars passed as args to 'vagrant up'
 echo Args passed: [[ $@ ]]
+
+echo -n "$1" > /etc/hostname
+hostname -F /etc/hostname
+
+/sbin/ip addr add "$3/24" dev eth1
+/sbin/ip link set eth1 up
+/sbin/ip link set eth2 up
+
 echo 'export GOPATH=#{netplugin_synced_gopath}' > /etc/profile.d/envvar.sh
 echo 'export GOBIN=$GOPATH/bin' >> /etc/profile.d/envvar.sh
 echo 'export GOSRC=$GOPATH/src' >> /etc/profile.d/envvar.sh
-echo 'export GOROOT=#{host_goroot_path}' >> /etc/profile.d/envvar.sh
-echo 'export PATH=$PATH:#{host_goroot_path}/bin:#{host_gobin_path}:$GOBIN' >> /etc/profile.d/envvar.sh
-if [ $# -gt 0 ]; then
-    echo "export $@" >> /etc/profile.d/envvar.sh
```

Comment:
The systemd stuff is very finicky (e.g., not even putting it into a shell profile and telling systemd to source it works) and was really hard to get right. I'd prefer to just leave it alone until someone really has a problem with it.